src/components/progress/progress.ts

```diff
@@ -0,0 +1,28 @@
+import { Component, Input, ViewEncapsulation, HostBinding } from '@angular/core';
+
+@Component({
+  selector: 'ion-progress',
+  templateUrl : './progress.html',
+  encapsulation: ViewEncapsulation.None
+})
+export class Progress {
+
+  constructor( ) { }
+
+  /**
+   * @input {'determinate' | 'indeterminate' | 'buffer' | 'query'} The current mode of the progress indicator
```

Comment:
@AmitMY I believe `query` is both determinate and indeterminate. It's been awhile since I looked at these but if I recall correctly it displays as `indeterminate` during the call for data and then `determinate` once it gets the data and is still loading. See here:

https://material.io/guidelines/components/progress-activity.html#progress-activity-types-of-indicators

I like using the name `query`, it's short and goes with the MD guidelines. Maybe we just need to document what each indicator does really well.